rusoto/core/src/region.rs

```diff
@@ -28,6 +28,7 @@ pub enum Region {
     UsWest1,
     UsWest2,
     CnNorth1,
+    Local(&'static str)
```

Comment:
I wonder if we do want to resolve all unknown region names as this `Local` region and let it fail at runtime? Compile-time checks are better obviously, but I can envision a team having a configuration file that differs between dev and prod that contains the region name in it. It would be useful to be able to automatically resolve that.

_However_, that being said, right now it's opt-in behaviour. If it fails to parse, the user can always take that value and put it in a `Region::Local(...)` themselves if they're sure about it.